### ISSUE TITLE
changing sample_rate gives unexpected result

### ISSUE THREAD

**Issue by seaniedan** [yistLin/FragmentVC#8]
I know this is an error in my thinking rather than your code, which is excellent, but I cannot figure out how to produce a good sounding sample at higher sample rates. I first tried (source and target wavs are sample rate 48000):
`convert.py -w downloaded_models/wav2vec_small.pt -v downloaded_models/vocoder.pt -c 'downloaded_models/fragmentvc.pt' source.wav target1.wav target2.wav --sample_rate 48000`

but the pitch was 3x too high, so I downloaded the VCTK data, preprocess at 48000:
`python preprocess.py ../../in/VCTK/ downloaded_models/wav2vec_small.pt features  --sample_rate 48000`

and retrained:
`python train.py features --batch_size 2 --save_dir ./ckpts`

...but even using the resulting model, the pitch is 3x higher than expected. Can you point me in the right direction?

EDIT: I noticed the speed AND pitch are too high:

$ mediainfo '/mnt/data/ml/sound/work/FragmentVC/out/Tom03.wav' 
General
Complete name                            : /mnt/data/ml/sound/work/FragmentVC/out/Tom03.wav
Format                                   : Wave
File size                                : 340 KiB
**Duration                                 : 3 s 629 ms**
Overall bit rate mode                    : Constant
Overall bit rate                         : 768 kb/s

Audio
Format                                   : PCM
Format settings                          : Little / Signed
Codec ID                                 : 1
Duration                                 : 3 s 629 ms
Bit rate mode                            : Constant
Bit rate                                 : 768 kb/s
Channel(s)                               : 1 channel
Sampling rate                            : 48.0 kHz
Bit depth                                : 16 bits
Stream size                              : 340 KiB (100%)

$ mediainfo '/mnt/data/ml/sound/in/Tom03.wav' 
General
Complete name                            : /mnt/data/ml/sound/in/Tom03.wav
Format                                   : Wave
File size                                : 608 KiB
Duration                                 : 6 s 412 ms
Overall bit rate mode                    : Constant
Overall bit rate                         : 777 kb/s
Producer                                 : Pro Tools
Description                              : dUBITS=$17082011 / dSCENE=25-4 / dTAKE=2 / dTAPE=IBDM10 / dFRAMERATE=25.000ND / dSPEED=025.000-NDF / dTRK2=Boom 1
Encoded date                             : 2020-08-17 00:00:00
Producer_Reference                       : aaymNFNMQkQk
UMID                                     : 0x60A2B340101010501010F1013000000A2AA9A77E7318000169C8EC3650D1B1C

Audio
Format                                   : PCM
Format settings                          : Little / Signed
Codec ID                                 : 1
**Duration                                 : 6 s 412 ms**
Bit rate mode                            : Constant
Bit rate                                 : 768 kb/s
Channel(s)                               : 1 channel
Sampling rate                            : 48.0 kHz
Bit depth                                : 16 bits
Stream size                              : 601 KiB (99%)

**Comment by yistLin** [comment]
If you preprocessed the training data, configured and retrained the model correctly, the model is then capable of generating Mel spectrogram that can possibly be inverted to 48KHz waveform by a vocoder.

However, the pretrained vocoder I provided can only generate 16KHz waveform. So if you wish to generate 48KHz audio, you'll have to retrain the vocoder as well.

**Comment by seaniedan** [comment]
Ah, I thought that might be the case. So now I've downloaded your universal-vocoder and I'm trying:
`python preprocess.py /data/sound/in/VCTK preprocessed --sample_rate 48000`
..and then I'll train the vocoder model on that!
Yist, thank you for being so clear with your instructions and generous with your time. I'll let you know how I get on. Let me know if you'd like me to send you a trained model.

**Comment by seaniedan** [comment]
Update: I trained the data with 
`python train.py preprocessed`
and ran a convert with
`convert.py -w downloaded_models/wav2vec_small.pt -v /universal-vocoder-trained-at-48000/vocoder-ckpt-100000.pt -c 'ckpts/retriever-best-loss0dot6863.pt' source.wav target1.wav target2.wav --sample_rate 48000`

..the output is now the right pitch, but it is twice the speed of the source.wav.

The vocoder model and checkpoint file 'ckpts/retriever-best-loss0dot6863.pt' were trained at 48000Hz. 

 I would be very grateful for your thoughts on how to proceed.

**Comment by swel4ik** [comment]
> Update: I trained the data with
> `python train.py preprocessed`
> and ran a convert with
> `convert.py -w downloaded_models/wav2vec_small.pt -v /universal-vocoder-trained-at-48000/vocoder-ckpt-100000.pt -c 'ckpts/retriever-best-loss0dot6863.pt' source.wav target1.wav target2.wav --sample_rate 48000`
> 
> ..the output is now the right pitch, but it is twice the speed of the source.wav.
> 
> The vocoder model and checkpoint file 'ckpts/retriever-best-loss0dot6863.pt' were trained at 48000Hz.
> 
> I would be very grateful for your thoughts on how to proceed.

Hi! Can you share your results, please? For smaple_rate=48k and 16k. For example with sound cloud links

**Comment by seaniedan** [comment]
[Here are the original and vocoded files at 48k.](https://soundcloud.com/sean-danischevsky/sets/vocoder-problems) I noticed the loss on vocoder training is quite high and doesn't reduce much after 100000 steps. It's around 2.3. Loss on the 'checkpoint path' model is .6863.

**Comment by swel4ik** [comment]
> [Here are the original and vocoded files at 48k.](https://soundcloud.com/sean-danischevsky/sets/vocoder-problems) I noticed the loss on vocoder training is quite high and doesn't reduce much after 100000 steps. It's around 2.3. Loss on the 'checkpoint path' model is .6863.

Thank you! I think that it is possible to try to change other parameters like: nfft, hop_len, win_len, not only the sample rate. 
Here you can find thre recommendations for these params:
![image](https://user-images.githubusercontent.com/61386841/101468385-61749b00-3954-11eb-922d-cbf1cf4ee0d2.png)
I tried to do it, but without retraining vocoder, so the loss on fragmnentVC did not reduce lower than 1.76. Maybe u can get better result than me:)

**Comment by seaniedan** [comment]
I noticed the defaults for the FragmentVC and universal-vocoder models are different, so I'm currently retraining the vocoder model with the n_fft, hop_length and win_length the same as the FragmentVC defaults (except with both set to --sample_rate 48000. I'll let you know if it improves anything.

**Comment by seaniedan** [comment]
Setting the parameters when retraining the vocoder model to match the parameters in FragmentVC helped:;it's now the correct speed. However the quality of the output is still very low. I feel I should concentrate on training the vocoder, as the loss is so large (2.3). Is this what is causing the problem?  Can anyone help me?

**Comment by yistLin** [comment]
@seaniedan Since we've never experimented on generating 48KHz results, I can't tell you whether FragmentVC is capable of generating fine-grained spectrogram in such high sampling rate.

As for the vocoder, it's not clear whether this WaveRNN-based vocoder can generate high quality 48KHz audio. As far as I know, the [original paper](https://arxiv.org/abs/1811.06292) only experimented on generating 24KHz audio.
